app/views/spree/orders/_line_item.html.haml

```diff
@@ -25,8 +26,13 @@
         = line_item.unit_price_price_and_unit
     %td.text-center.cart-item-quantity{"data-hook" => "cart_item_quantity"}
       - finalized_quantity = @order.completed? ? line_item.quantity : 0
+      - max_quantity = variant.on_demand && 9999 || variant.on_hand
       = item_form.number_field :quantity,
-        :min => 0, "ofn-on-hand" => "#{variant.on_demand && 9999 || variant.on_hand}",
+        :min => 0,
+        :max => max_quantity,
+        "data-line-item-id" => line_item.id,
+        "data-action": "change->cart#onQuantityChange",
+        "ofn-on-hand" => "#{variant.on_demand && 9999 || variant.on_hand}",
```

Comment:
This line appears to no longer be used, can it be removed?

```suggestion
```

Or if still required, it could use the `max_quantity` variable.

Reply:
It seems to be used, at least through `app/assets/javascripts/darkswarm/directives/validate_stock_quantity.js.coffee`

Oh, for sure I need to use `max_quantity`, that's why I created a variable ;)

Reply:
Ah of course, I didn't think to look for the camelised version `ofnOnHand`.